platform-hub-api/app/services/new_user_service.rb

```diff
@@ -0,0 +1,28 @@
+class NewUserService
+
+  def create auth_token_payload
+    return nil if auth_token_payload.blank?
+
+    user = User.new
+
+    user.id = auth_token_payload['sub']
+    user.name = auth_token_payload['name']
+    user.email = auth_token_payload['email']
+
+    user.identities.build(
+      provider: 'keycloak',
+      external_id: auth_token_payload['sub'],
+      external_username: auth_token_payload['preferred_username'],
+      external_name: auth_token_payload['name'],
+      external_email: auth_token_payload['email'],
+      data: auth_token_payload
+    )
+
+    user.last_seen_at = DateTime.now
+
+    user.save!
```

Comment:
Yep – that's what I want :) Hoping that it's better to fail early and catch in dev/test. (Note: given it's an API endpoint, the frontend app will catch the failed API request and show and error message to the user).